tests/pytorch/test_datasets_pt.py

```diff
@@ -201,3 +201,37 @@ def test_charactergenerator():
     assert isinstance(images, torch.Tensor) and images.shape == (2, 3, *input_size)
     assert isinstance(targets, torch.Tensor) and targets.shape == (2,)
     assert targets.dtype == torch.int64
+
+
+@pytest.mark.parametrize(
+    "dataset_name, train, input_size, size, rotate",
+    [
+        ['SVHN', True, [32, 128], 3, True],  # Actual set has 33402 samples
+        ['SVHN', False, [32, 128], 3, False],  # Actual set has 13068 samples
+    ],
+)
+def test_mock_dataset(dataset_name, train, input_size, size, rotate, mock_svhn_dataset):
+    # monkeypatch the path to temporary dataset
+    if dataset_name.lower() == 'svhn':
+        dataset_path = mock_svhn_dataset
+
+    datasets.__dict__[dataset_name].MOCK_PATH = dataset_path
```

Comment:
```python
# monkeypatch the path to temporary dataset
datasets.SVHN.TRAIN = (mock_svhn_dataset, None, "svhn_train.tar")
datasets.SVHN.TEST = (mock_svhn_dataset, None, "svhn_test.tar")
```

Reply:
@fg-mindee 
does not work:
`E           ValueError: the dataset needs to be downloaded first with download=True`
this was my reason for the if else statements :sweat:

Reply:
This PR https://github.com/mindee/doctr/pull/694 should do the trick :ok_hand: 
But you'll have to pass `cache_dir` and `cache_subdir` in the good way so that it picks up your temporary files